src/tesla-notes.md

```diff
@@ -0,0 +1,460 @@
+${\bf A}$ boundary condition which can be used to apply an external
+magnetic field.
+
+There also exists a special case in magnetostatics when the current
+density is equal to zero. In this case $\nabla\times{\bf H}=0$
+which implies that the magnetic field can be computed as
+${\bf H}=-\nabla\Phi_M$. This leads to the scalar potential
+formulation which we will not consider further except to say that the
+electrostatic solver, named `volta`, can be adapted to model such
+situations.
+
+# The `tesla` Miniapp
+
+The `tesla` miniapp models the magnetostatic equation for the magnetic
+vector potential ${\bf A}$. It includes source terms derived from a
+volumetric current source ${\bf J}$, magnetization vector
+${\bf M}$, or surface currents ${\bf K}$.
+$$\nabla\times(\mu^{-1}\nabla\times{\bf A}) = {\bf J}+\nabla\times(\mu^{-1}\mu_0{\bf M})$$
+$$\hat{n}\times(\mu^{-1}\nabla\times{\bf A}) = \hat{n}\times{\bf K}$$
+
+The magnetic vector potential will be approximated in
+H(Curl) so that the left hand side operator is well defined.
+$${\bf A} \approx \sum_i a_i {\bf W}_i (\vec{x})$$
+
+Inserting this into the left hand side of the equation and integrating the resulting
+equation against each H(Curl) basis function leads to the following weak
+form:
+$$\begin{align}
+\int\_{\Omega}{\bf W}\_{i}(\vec{x})\cdot[\nabla\times(\mu^{-1}\nabla\times{\bf A})]d\Omega & \approx
+\int\_\Omega{\bf W}_i(\vec{x})\cdot\{\nabla\times[\mu^{-1}\nabla\times(\sum_j a_j{\bf W}_j(\vec{x}))]\}d\Omega \\\\
+  & = \sum\_j a_j\{\int\_\Omega{\bf W}_i(\vec{x})\cdot[\nabla\times(\mu^{-1}\nabla\times{\bf W}_j(\vec{x}))]d\Omega\}
+\end{align}$$
+
+
+The expression in curly braces depends only on our material coefficient and our basis functions. This particular integral
+requires a little more manipulation to move the outermost curl operator
+onto the H(Curl) basis function. $$\begin{aligned}
+  \int\_\Omega{\bf W}_i(\vec{x})\cdot[\nabla\times(\mu^{-1}\nabla\times{\bf W}_j(\vec{x}))]\,d\Omega
+  &=&
+  \int\_\Omega(\nabla\times{\bf W}_i(\vec{x}))\cdot(\mu^{-1}\nabla\times{\bf W}_j(\vec{x}))\,d\Omega \\\\
+  &-&  \int\_\Omega\nabla\cdot[{\bf W}_i(\vec{x})\times(\mu^{-1}\nabla\times{\bf W}_j(\vec{x}))]\,d\Omega \\\\
+  &=&
+  \int\_\Omega(\nabla\times{\bf W}_i(\vec{x}))\cdot(\mu^{-1}\nabla\times{\bf W}_j(\vec{x}))\,d\Omega \\\\
+  &-&  \int\_\Gamma\hat{n}\cdot[{\bf W}_i(\vec{x})\times(\mu^{-1}\nabla\times{\bf W}_j(\vec{x}))]\,d\Gamma \\\\
+  &=&
+  \int\_\Omega(\nabla\times{\bf W}_i(\vec{x}))\cdot(\mu^{-1}\nabla\times{\bf W}_j(\vec{x}))\,d\Omega \\\\
+  &+&  \int\_\Gamma{\bf W}_i(\vec{x})\cdot[\hat{n}\times(\mu^{-1}\nabla\times{\bf W}_j(\vec{x}))]\,d\Gamma \\\\
+\end{aligned}$$ The first integral remaining on the right hand side is
+implemented in MFEM as a `BilinearFormIntegrator` named
+`CurlCurlIntegrator`. The second integral, the boundary integral, gives
+rise to a Neumann boundary condition which will be discussed further in
+Section [2.1.3](#sec:surf_current){reference-type="ref"
```

Comment:
The portion of this reference in curly braces may be superfluous. These curly braces and their contents appear in my web browser. @acfisher, are they hidden in yours?

Reply:
Oops, I missed that one.  I hit the LaTeX file with a converter and had to do a bunch of hand editing.